### DISCUSSION TITLE
State of v7

### DISCUSSION THREAD

**Discussion by marcin-bury** [sequelize/sequelize#15736] [Help & Questions] [answered]
Hi

I'm new to TS/JS world, although being experienced developer.
I wondering if v7 is mature enough to start learning how to use it. I prefer TS over JS, and to be honest I'd like to skip 'sequelize-typescript' bridge.

Thx
Marcin

**Comment by ephys** [comment, marked as answer]
Sequelize 7 is still in alpha but if you're OK with the limitations, it's not a bad idea to start using it.

One of the selling points of v7 is that we rewrote a big part of it to be typescript-friendly

You can take a look at the progress of Sequelize 7 here: https://github.com/orgs/sequelize/projects/1/views/1

Here is what is likely to impact you:
- sequelize-cli does not support Sequelize 7 yet. If you want to use migrations, you would need to use umzug or another migration library instead
- If you use third-party libraries, they are unlikely to work with Sequelize 7 for the time being
- ~A few decorators from Sequelize-TypeScript are still missing: https://github.com/sequelize/sequelize/issues/15334~
- ~Model inheritance is not ready~ Done
- ~There is a minor bug with belongsTo (with workaround): https://github.com/sequelize/sequelize/issues/15625, which will be fixed before the stable release~. Fixed
- ~The new documentation does not describe how to use association decorators yet. They work very similarly to how they did in Sequelize-TypeScript~. Done

~If you want to use Sequelize 7, I recommend reading the documentation from this PR: https://github.com/sequelize/website/pull/330#issuecomment-1445352867.~ The documentation is available on https://sequelize.org.  Don't hesitate to ask questions as it will help us improve the documentation

**Reply by marcin-bury** [reply]
Zoe

Thank you very much for the detailed answer.

Marcin

**Comment by taslangraham** [comment]
@ephys Will sequelize-cli support Sequelize 7 anytime soon?

**Reply by ephys** [reply]
We don't have an ETA at this time. We'll work on the CLI when the main library is ready

**Comment by ilyaskarim** [comment]
@ephys When you think the Sequelize 7 will be in stable version....

**Reply by ephys** [reply]
We discussed it during last month's meeting, the meeting notes are here: https://github.com/sequelize/meetings/wiki/Meeting-2024%E2%80%9003%E2%80%9022#latest-v7-release-plan

The last two blockers that remained were splitting the dialects ([this is now done](https://github.com/sequelize/sequelize/releases/tag/v7.0.0-alpha.40)) and implementing the new CLI, ([this is in progress](https://github.com/sequelize/sequelize/tree/main/packages/cli))

**Reply by ilyaskarim** [reply]
Thank you @ephys for the links those were helpful, BTW I am very happy with decision of including sorting in each include for associations.

**Comment by ilyaskarim** [comment]
@ephys another question, is there any planning to include support for Sequelize 7 in `sequelize-auto` package?

**Reply by WikiRik** [reply]
The current plan is that both the features of `sequelize-auto` and `sequelize-cli` will be released as `@sequelize/cli`. The features of the current packages will not completely be migrated when 7.0.0 releases and we will keep working on them after the initial release

**Reply by ilyaskarim** [reply]
Cool thanks @WikiRik

**Comment by mschipperheyn** [comment]
Is June still the objective for beta release?

**Comment by rt-joe** [comment]
Any recent updates?

**Comment by jackhannon** [comment]
What is the status of this?

**Comment by slidenerd** [comment]
with all due respect to all the maintainers involved, you guys really need to get it together. projects like drizzle and prisma are gaining big time over you guys thanks to the speed at which they are moving. Using sequelize with typescript is a total nightmare adding so many libraries that don't even work with all the features well. For example, [many to many is still broken](https://github.com/sequelize/sequelize-typescript/issues/1316)

**Comment by ryansuhartanto** [comment]
For anyone blocked on migrations while `@sequelize/cli` is still in progress: I built https://github.com/ryansuhartanto/sqlumz as a stopgap. It's a small CLI on top of umzug for v7 migrations and seeds, in plain SQL or TS/JS, with `run`/`undo`/`status` and `--to`/`--step`. Happy to retire it once the official CLI ships.
